tests/protractor/upload/upload.js

```diff
@@ -0,0 +1,39 @@
+var app = angular.module('upload', ['firebase.storage']);
+
+app.controller('UploadCtrl', function Upload($scope, $firebaseStorage, $timeout) {
+  // Create a reference (possible create a provider)
+  const storageRef = firebase.storage().ref('user/1.png');
+  // Create the storage binding
+  const storageFire = $firebaseStorage(storageRef);
+
+  var file;
+
+  $scope.select = function (event) {
+    file = event.files[0];
+  }
+
+  $scope.upload = function() {
+    $scope.isUploading = true;
+    $scope.metadata = {bytesTransferred: 0, totalBytes: 1};
+    $scope.error = {};
```

Comment:
Also, we never actually use `$scope.error` in the HTML file. We should add a conditional `<div>` which shows the error if there is one.